SeaPublicWebsite.UnitTests/CookieServiceTests.cs

```diff
@@ -0,0 +1,247 @@
+        var response = context.Response;
+        request.Path = "/cookies";
+
+        // Act
+        var bannerState = CookieService.GetAndUpdateBannerState(request, response);
+        
+        // Assert
+        bannerState.Should().Be(BannerState.Hide);
+    }
+    
+    [Theory]
+    public void ShowsBannerIfSettingsAreOutdatedOrMissing(CookieSettings value)
+    {
+        // Arrange
+        var context = new DefaultHttpContext();
+        var request = context.Request;
+        var response = context.Response;
+        request.Headers.Cookie = $"{Key}={ConvertObjectToHttpHeaderSrting(value)}";
+        
+        // Precondition
+        Assume.That(!CookieService.CookieSettingsAreUpToDate(request));
+
+        // Act
+        var bannerState = CookieService.GetAndUpdateBannerState(request, response);
+        
+        // Assert
+        bannerState.Should().Be(BannerState.ShowBanner);
+    }
+    
+    [Theory]
+    public void HidesBannerIfCookiesWereSetAndConfirmationWasShown(CookieSettings value)
+    {
+        // Arrange
+        var context = new DefaultHttpContext();
+        var request = context.Request;
+        var response = context.Response;
+        request.Headers.Cookie = $"{Key}={ConvertObjectToHttpHeaderSrting(value)}";
+        
+        // Precondition
+        Assume.That(CookieService.CookieSettingsAreUpToDate(request));
+        Assume.That(value.ConfirmationShown);
+
+        // Act
+        var bannerState = CookieService.GetAndUpdateBannerState(request, response);
+        
+        // Assert
+        bannerState.Should().Be(BannerState.Hide);
+    }
+    
+    [Theory]
+    public void ShowsConfirmationBannerAndUpdatesRequestCookieIfItWasNotShownAlready(CookieSettings value)
+    {
+        // Arrange
+        var context = new DefaultHttpContext();
+        var request = context.Request;
+        var response = context.Response;
+        request.Headers.Cookie = $"{Key}={ConvertObjectToHttpHeaderSrting(value)}";
+        
+        // Precondition
+        Assume.That(CookieService.CookieSettingsAreUpToDate(request));
+        Assume.That(!value.ConfirmationShown);
+
+        // Act
+        var bannerState = CookieService.GetAndUpdateBannerState(request, response);
+        
+        // Assert
+        var expectedBannerState = value.GoogleAnalytics ? BannerState.ShowAccepted : BannerState.ShowRejected;
+        bannerState.Should().Be(expectedBannerState);
+        value.ConfirmationShown = true;
+        AssertResponseContainsCookie(response, Key, value);
+    }
+
+    private void AssertResponseContainsCookie(HttpResponse response, string key, object value)
+    {
+        response.Headers.SetCookie.ToString().Should().Contain($"{key}={ConvertObjectToHttpHeaderSrting(value)}");
+    }
+
+    private string ConvertObjectToHttpHeaderSrting(object o)
+    {
+        return Uri.EscapeDataString(JsonConvert.SerializeObject(o));
+    }
+
+    private CookieSettings[] SetUpCookieSettings()
```

Comment:
This could just be
```
[DatapointSource] 
    private CookieSettings[] CookieSettings = {
        new CookieSettings // Accepted latest cookies
        {
            Version = LatestVersion,
            ConfirmationShown = true,
            GoogleAnalytics = true
        },
        new CookieSettings // Outdated version
        {
            Version = LatestVersion - 1,
            ConfirmationShown = true,
            GoogleAnalytics = true
        },
        ...
};
```

Reply:
More than this, I also needed to create a test case class to add a toString method so that failing cases have a name